Python/Source/RandomPick.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Rover/Evaluator.hpp"
+#include "Rover/Range.hpp"
+#include "Pick.hpp"
+#include "RandomPick.hpp"
+
+using namespace pybind11;
+using namespace Rover;
+
+namespace {
+  class PythonRandomPick {
+    public:
+      class Choice {
+        public:
+          explicit Choice(std::size_t num_generators)
+            : m_range(0, num_generators - 1) {}
+
+          std::size_t operator ()() {
+            auto evaluator = Evaluator();
+            return m_range.generate(evaluator);
+          }
+
+        private:
+          Range<std::size_t, std::size_t> m_range;
+      };
+      
+      using Type = object;
+
+      PythonRandomPick(args generators)
+        : m_pick(cast(Choice(generators.size())), std::move(generators)) {}
+
+      Type generate(Evaluator& evaluator) {
+        return m_pick.generate(evaluator);
+      }
+
+    private:
+      PythonPick m_pick;
+  };
+}
+
+void Rover::export_random_pick(module& module) {
+  class_<PythonRandomPick::Choice>(module, "Choice")
```

Comment:
I don't think you need to export ```Choice```, and if you do, then it needs to be nested within ```RandomPick``` rather than part of the module's namespace.

Reply:
I had to export `Choice` here because I cast it to `pybind11::object` to re-use `PythonPick`. I can avoid that, but then I cannot use `PythonPick` as it works with Python functions. However, all that may be changed slightly once I make `Choice` a generator (which I will export anyway).